### ISSUE TITLE
Install Process Fails at Update Stage (late 2013 iMac)

### ISSUE THREAD

**Issue by adamhenderson** [BenSova/Patched-Sur#113]
**Description**
Went through the steps described for Patched Sur, however Big Sur installation failed at a late stage when trying to perform an update as part of the install process.

**To Reproduce**
1. Following running and setting up USB stick with Patched Sur and rebooting to startup disk selection (Wifi connected)
2. Select Install Big Sur and selecting target drive
3. Installation happens but upon automatic restart into Big Sur Installation screen an error modal saying "There was an issue updating the installation' (paraphrased) 

install-log excerpt:
Dec  3 08:30:45 iMac OSInstaller[199]: ------- Install Failed -------
Dec  3 08:30:45 iMac OSInstaller[199]: Operation: Install Mobile Software Update failed, Failure Reason: Error Domain=com.apple.osinstall Code=-83 "An error occurred applying the software update." UserInfo={NSLocalizedDescription=An error occurred applying the software update., NSUnderlyingError=0x7f9162d6d1b0 {Error Domain=MobileSoftwareUpdateErrorDomain Code=29 "Failed to verify package contents." UserInfo={NSLocalizedDescription=Failed to verify package contents., NSUnderlyingError=0x7f9162d6d160 {Error Domain=MobileSoftwareUpdateErrorDomain Code=7 "SHA-1 mismatch for /Volumes/Update/softwareupdate.279.dz3VsC/source/payloadv2/fixup.manifest: 0b0f4638e8b61637e62b022a0f4f19c7848da938 should be d783067e5c7854d9334911adaa97b80cf65ac04f" UserInfo={NSLocalizedDescription=SHA-1 mismatch for /Volumes/Update/softwareupdate.279.dz3VsC/source/payloadv2/fixup.manifest: 0b0f4638e8b61637e62b022a0f4f19c7848da938 should be d783067e5c7854d9334911adaa97b80cf65ac04f, target_update=20B50}}}}}

**Expected behavior**
As per instructions, installation should complete and boot fully into Big Sur

**Desktop (please complete the following information):**
 - Upgrading from: Catalina v10.15.7
- Upgrading to: Big Sur v11.0.1 (latest)
 - Mac Model: Late 2013 iMac 27"
 - Patched Sur Version v0.0.4

**Comment by iMiMx** [comment]
Did you do the below, selecting 'EFI Boot', before selecting 'Install Big Sur'?  It was 'red' not yellow on mine, however.

> 11. Select the yellow EFI Boot drive, (if there are multiple unplug and replug in your drive and select the EFI Boot that disappeared and reappeared). Then, your Mac will quickly turn off, so turn it back on while, again, holding down Option/Alt. Then, select Install macOS Big Sur Beta.


Aside from the above, could it be a dodgy USB stick?  

```
Failed to verify package contents.
SHA-1 mismatch for 
```

I just did the same upgrade last night, Late 2013 iMac 27 as well.

**Comment by adamhenderson** [comment]
Yes I did the EFI boot - was red not yellow. 
I used a brand new 32GB Kingston stick so I would doubt it was the stick. 
Will have another stab at it - I’m practicing on a spare Catalina drive before I commit to doing it for real on my real boot drive.

**Comment by iMiMx** [comment]
> Will have another stab at it - I’m practicing on a spare Catalina drive before I commit to doing it for real on my real boot drive.

I wonder if this is part of the problem, I'm sure I read in another post that if the drive names change, the stick has to be recreated - perhaps the drive IDs are changing.

**Comment by adamhenderson** [comment]
Yeah maybe. To be honest this Mac of mine is on its last legs, the HDD part of the fusion drive has failed so I split the fusion drive so the SSD is a separate independent drive with Catalina on and that is what I’m trying to upgrade.

**Comment by adamhenderson** [comment]
So I tried again, multiple restarts but always ends in failure at this point

![A759DB4F-F4F0-4C00-AA9F-BD737CC9A936](https://user-images.githubusercontent.com/6618734/101067409-6d94dd00-358f-11eb-8c5d-f44ea0d668ab.jpeg)

**Comment by adamhenderson** [comment]
Success! I zapped the pram before reattempting the install with Command, Option , P & R.
Steps:
Zap Pram
Reboot to Disk Selection (alt) (usb stick already created and plugged in)
Select Red EFI (quickly shuts down after selection)
Reboot to Disk Selection (alt)
Select Green Install Big Sur
Follow installer selecting target drive
Restarted into Big Sur!

**Comment by iMiMx** [comment]
Cool!  I got an error when trying to perform the post install tasks, similar to the below:

> nvram: Error getting variable - 'csr-active-config': (iokit/common) data was not found
> csr-active-config appears to be set incorrectly:
> nvram: Error getting variable - 'csr-active-config': (iokit/common) data was not found
> 
> To fix this, please boot the setvars EFI utility, then boot back into macOS
> and try again. Or if you believe you are seeing this message in error, try the
> `--force` command line option.

When running it from the CLI with --force ( /Volumes/Install\ macOS\ Big\ Sur/patch-kexts.sh) it actually spits out a better error:

> No WiFi option specified on command line, so checking for 802.11ac...
> Found 802.11ac WiFi card, so not installing a WiFi patch.
> No patch mode specified on command line. Detecting Mac model...
> (Use --2010, --2011, or --2012 command line option to override.)
> Detected model: iMac14,2
> Detected a Late 2013 iMac. patch-kexts.sh is not necessary on this model.

Which confirms how ridiculous not supporting the Late 2013 iMac is anyway... nice to know that hackery needs to be kept to a minimum anyway!

**Comment by adamhenderson** [comment]
Yes - its bizarre these machines are not supported officially and I'm very impressed with what Ben an others have done to make this possible in a (relatively) easy way - great work!

**Comment by adamhenderson** [comment]
Update. 
I managed to get Big Sur installed on the internal SSD but doing an upgrade from Catalina on my main boot external USB has been completely unsuccessful so far. I booted to the green Install Big Sur from my Patched Sur usb stick and went through the BS install selecting my target external drive, this just results in the black Apple progress bar screen - it gets to about 13mins remaining and it just reboots and gets into a loop. 
Have tried NVRAM resets, disk first aid, nothing seems to get past this loop. 
Running out of ideas.

**Comment by ssthakran** [comment]
> Success! I zapped the pram before reattempting the install with Command, Option , P & R.
> Steps:
> Zap Pram
> Reboot to Disk Selection (alt) (usb stick already created and plugged in)
> Select Red EFI (quickly shuts down after selection)
> Reboot to Disk Selection (alt)
> Select Green Install Big Sur
> Follow installer selecting target drive
> Restarted into Big Sur!

Good Morning All,
I Bought a New 32 GB Pen Drive , Zapped the Pram , Downloaded Both Patched Sur and Install Package
Put them on the Desktop
And Followed the Steps
Took a While But
Big Sur Up and Running 
INFACT I DID NOT HAVE TO EVEN LOAD THE KEXTS FILE OR DISABLE ANIMATION >>>>
Everything is Working like Clockwork ....
THANKS A TON TO EVERYONE INVOLVED 
GREAT JOB AND WELL DONE
HAVE A GREAT DAY

**Comment by ssthakran** [comment]
iMac Late 2013
Updated from Catalina 10.15.7

**Comment by CK0y0TE** [comment]
As it seems Apple decided to drop the late 2013 in a later stage.
Mine too 14,3 model completely supported without "post install".
(no fusion drive, replaced HDD with SSD Sata a year ago. core i7, NVidia 750M 1Gb)
clockwork. Very happy.
